toolchain/parse/handle_decl_scope_loop.cpp

```diff
@@ -25,81 +67,192 @@ static auto OutputInvalidParseSubtree(Context& context, int32_t subtree_start)
                   /*has_error=*/true);
 }
 
-// Handles an unrecognized declaration.
-static auto HandleUnrecognizedDecl(Context& context, int32_t subtree_start)
-    -> void {
+// Reports an error and outputs an invalid parse tree.
```

Comment:
I had trouble understanding what "outputs an invalid parse tree" means.

It seems like the purpose of `OutputInvalidParseSubtree` is to recover from an error by skipping to a good stopping place and fixing up the parse tree to match. I don't get that from the name. Maybe something more like `FinishAndSkipInvalidDecl` would be clearer?